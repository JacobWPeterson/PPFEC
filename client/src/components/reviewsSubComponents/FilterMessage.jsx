import React from 'react';
import styled from 'styled-components';

const ClickTag = styled.div`
  padding: 2px 1vw 0 1vw;
  text-decoration:underline;

  &.not-click {
    &:hover {
      cursor: pointer;
      color: #80CCC4;
      transform: scale(1.1);
    }
  }
`;

// const FilterMessage = (props) => {
//   return (

//     <div>
//       <span>Showing</span>
//       {prpos.filter.map((star) => (
//         <span>{star} stars reviews</span>
//       ))}
//     </div>
//     <ClickTag>
//       Remove All Filters
//     </ClickTag>
//   )
// }

export default FilterMessage;
