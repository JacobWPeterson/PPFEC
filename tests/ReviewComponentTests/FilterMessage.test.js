/* eslint-disable react/jsx-props-no-spreading */
import React from 'react';
import { mount } from 'enzyme';
import FilterMessage from '../../client/src/components/reviewsSubComponents/FilterMessage.jsx';

describe('FilterMessage Component Test', () => {
  const props = {
    filterStars: [1, 2, 4],
  };

  let FilterMessageComponent;
  beforeEach(() => {
    FilterMessageComponent = mount(<FilterMessage {...props} debug />);
  });

  afterEach(() => {
    jest.clearAllMocks();
  });

  it('Should render correctly', () => {
    // mount render the whole dom with styled-component
    expect(FilterMessageComponent).toMatchSnapshot();
  });
});