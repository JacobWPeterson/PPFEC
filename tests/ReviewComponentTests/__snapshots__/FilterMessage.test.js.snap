// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilterMessage Component Test Should render correctly 1`] = `
<FilterMessage
  debug={true}
  filterStars={
    Array [
      1,
      2,
      4,
    ]
  }
>
  <FilterMessage__FilterMessageContainer>
    <div
      className="FilterMessage__FilterMessageContainer-pnsqsl-1 iypfKb"
    >
      <div>
        <span>
          Showing all 
        </span>
        <span
          key="1"
        >
          1
        </span>
        <span
          key="2"
        >
          ,
          2
        </span>
        <span
          key="4"
        >
           
          &
          4
        </span>
        <span>
           star reviews
        </span>
      </div>
      <FilterMessage__ClickTag>
        <div
          className="FilterMessage__ClickTag-pnsqsl-0 cuCkFY"
        >
          Remove All Filters
        </div>
      </FilterMessage__ClickTag>
    </div>
  </FilterMessage__FilterMessageContainer>
</FilterMessage>
`;
